src/screens/ContactScreen/index.js

```diff
@@ -113,70 +102,22 @@ const ContactScreen = ({navigation}) => {
       setIsRecyclerViewShown(true);
     }
   }, [dataProvider]);
-
   const handleCreateChannel = async () => {
     try {
-      if (followed.length < 1) {
-        Alert.alert('Warning', 'Please choose min one user');
-      }
-      setLoading(true);
-      const members = followed;
-      const channelName = usernames;
-      let typeChannel = 0;
-      if (members.length > 2) {
-        typeChannel = 1;
+      const mappingUserName = selectedUsers?.map((user) => user?.username).join(',');
+      let image = DEFAULT_PROFILE_PIC_PATH;
+      if (selectedUsers.length === 1) {
+        image = selectedUsers[0]?.profile_pic_path;
       }
-      const clientChat = await client.client;
-
-      let type = 'messaging';
-      if (members.length > 2) {
-        type = 'group';
-      }
-      const filter = {type, members: {$eq: members}};
-      const sort = [{last_message_at: -1}];
-      const findChannels = await clientChat.queryChannels(filter, sort, {
-        watch: true,
-        state: true
-      });
-
-      const generatedChannelId = generateRandomId();
-      const memberWithRoles = members.map((item) => ({
-        user_id: item,
-        channel_role: 'channel_moderator'
-      }));
-
-      if (findChannels.length > 0) {
-        setChannel(findChannels[0], dispatchChannel);
-      } else {
-        const channelChat = await clientChat.channel(type, generatedChannelId, {
-          name: channelName.join(', '),
-          type_channel: typeChannel
-        });
-        await channelChat.create();
-        channelChat.update(
-          {
-            name: channelName.join(', ')
-          },
-          {
-            text: 'You created this group',
-            system_user: profile?.myProfile?.user_id,
-            is_from_prepopulated: true,
-            other_text: `${profile?.myProfile?.username} created this group`
-          }
-        );
-        await channelChat.addMembers(memberWithRoles);
-        setChannel(channelChat, dispatchChannel);
-      }
-      setFollowed([profile.user_id]);
-      setUsernames([profile.username]);
-      setLoading(false);
-      await navigation.replace('ChatDetailPage');
-    } catch (error) {
-      showMessage({
-        message: 'Failed creating new chat',
-        type: 'danger'
-      });
-      setLoading(false);
+      const dataSelected = {
+        user: {
+          name: mappingUserName,
+          image
+        }
+      };
+      createSignChat(followed, dataSelected);
+    } catch (e) {
+      console.log(e, 'error signed chat');
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new implementation of `handleCreateChannel` function seems to be simpler than the previous one, but it lacks some checks and error handling present in the old code. For instance, there's no check for the number of selected users before creating a chat. Also, the error handling only consists of a console log statement which might not be sufficient for production code. Consider adding more robust error handling and user input validation.
```diff
+    const handleCreateChannel = async () => {
+      try {
+        if (selectedUsers.length < 1) {
+          Alert.alert('Warning', 'Please choose min one user');
+          return;
+        }
+        const mappingUserName = selectedUsers?.map((user) => user?.username).join(',');
+        let image = DEFAULT_PROFILE_PIC_PATH;
+        if (selectedUsers.length === 1) {
+          image = selectedUsers[0]?.profile_pic_path;
+        }
+        const dataSelected = {
+          user: {
+            name: mappingUserName,
+            image
+          }
+        };
+        createSignChat(followed, dataSelected);
+      } catch (e) {
+        // Consider using a more user-friendly way to display errors, like an alert or a toast message
+        Alert.alert('Error', 'Failed to create signed chat');
+      }
+    };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->